Add creating snapshots for a single C file

PLukas2018 opened the pull request:
This PR extends `diffkemp build` to be able to make snapshots from a single C file.

PLukas2018 commented:
I used the same clang flags that are used by `diffkemp build` now
https://github.com/viktormalik/diffkemp/blob/2db77b5759bd5b3f889d078af2932a2db1affbd8/diffkemp/building/cc_wrapper.py#L176 

but should not it be somehow unified with flags that are used for `diffkemp build-kernel`
https://github.com/viktormalik/diffkemp/blob/2db77b5759bd5b3f889d078af2932a2db1affbd8/diffkemp/llvm_ir/kernel_llvm_source_builder.py#L347-L348
?

viktormalik commented on diffkemp/llvm_ir/wrapper_build_finder.py:
```diff
@@ -38,16 +38,7 @@ def initialize(self):
         # Read LLVM IR files, get all LLVM functions that satisfy C naming from
         # there and save them into self.functions along with the files there
         # were found in
```
This comment is not relevant anymore, let's remove it from here.

viktormalik commented on README.md:
```diff
@@ -92,6 +92,10 @@ DiffKemp runs in two phases:
     also has additional options to configure the project build, see `diffkemp
     build --help` for the complete list.
 
+    The command can be also used for a snapshot generation for a single C file,
+    in this case, the `PROJ_DIR` represents the path to the file. 
+    If `SYMBOL_LIST` is not given, all functions from the file are considered.
```
Small rewording:
```suggestion
    The command can be also used to generate a snapshot from a single C file.
    In this case, the path to the file should be given in place of `PROJ_DIR`. 
    
    If `SYMBOL_LIST` is not given, all functions from the project/file are considered.
```

PLukas2018 replied:
I omitted the last line because `SYMBOL_LIST` is already mentioned in the lines above
https://github.com/viktormalik/diffkemp/blob/abab4e4bdf8ec485d944a2d8b552a2a736dff257/README.md?plain=1#L85-L86

viktormalik commented on diffkemp/llvm_ir/single_c_builder.py:
```diff
@@ -0,0 +1,42 @@
+        :param clang: clang compiler to be used
+        :param clang_append: list of args to add when compiling
+        """
+        llvm_file_name = c_file_name.rsplit(".", 1)[0] + ".ll"
```
Use `os.path.splitext[0] + .ll`.

viktormalik commented on diffkemp/llvm_ir/single_c_builder.py:
```diff
@@ -0,0 +1,42 @@
+        :param clang: clang compiler to be used
+        :param clang_append: list of args to add when compiling
+        """
+        llvm_file_name = c_file_name.rsplit(".", 1)[0] + ".ll"
```
Also shouldn't be?
```suggestion
        self.llvm_file_name = c_file_name.rsplit(".", 1)[0] + ".ll"
```

viktormalik replied:
I missed that, correct.

viktormalik reviewed (commented):
Can we add a unit test for this?

viktormalik commented:
> I used the same clang flags that are used by `diffkemp build` now
>
> but should not it be somehow unified with flags that are used for `diffkemp build-kernel`
> ?

Good point. The `build-kernel` args were chosen like this b/c we wanted to have control over what LLVM passes are being applied. I agree that we should do the same for `build` but that's a different discussion from what this PR is about so let's keep it as it is for now.

PLukas2018 commented:
I addressed the comments, fixed a mistake I made (returning a dict instead of a list) and added tests. I guess I could add more tests (testing `SYMBOL_LIST` or `build` command in general...).

viktormalik commented:
Thanks! Can you squash the commits? Looks like leaving only the first two is the best.

viktormalik reviewed (approved):
Works nicely, thanks!
